### PR TITLE
chore: add AI agents, prompts, skills, and instructions for kvx

### PR DESCRIPTION
**Pull request by abaker-9**
Add Copilot customization files adapted from scafctl for kvx:
- 6 agents: commit-message, go-build-resolver, go-reviewer, issue-creator, planner, pr-reviewer
- 8 prompts: /commit, /completeness-check, /go-build, /go-review, /go-test, /issue, /plan, /pr-review
- 2 skills: golang-patterns, golang-testing
- 6 instructions: cli-commands, docs-examples, go-conventions, go-modules, go-testing, markdown
- All references updated from scafctl to kvx architecture (core engine, TUI, navigator, loader, formatter, CEL)
- Removed scafctl-only files: mcp-tools, providers, integration-tests instructions

### CODE REVIEW

File: .github/agents/pr-reviewer.agent.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+### Phase 5: Respond & Resolve
+
+**Only after all fixes pass verification**, respond to review threads:
+
+**To reply to a thread:**
+```bash
+gh api graphql -f query='
+  mutation($threadId: ID!, $body: String!) {
+    addPullRequestReviewThread(input: {pullRequestReviewThreadId: $threadId, body: $body}) {
+      comment { id }
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/agents/pr-reviewer.agent.md]: Fixed -- updated mutation to addPullRequestReviewThreadReply. Thanks!

File: .github/agents/go-build-resolver.agent.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  - label: "Generate commit message"
+    prompt: "Generate a commit message for the fixes just applied."
+    agent: "commit-message"
+---
+You are an expert Go build error resolution specialist for the **kvx** project. Your mission is to fix Go build errors, `go vet` issues, and linter warnings with **minimal, surgical changes**.
+
+## Project Context
+
+- kvx is a terminal-based UI for exploring structured data (JSON, YAML, TOML, NDJSON, CSV)
+- Build: `go build -ldflags "-s -w -X main.BuildTime=$(date -u +%Y-%m-%dT%H:%M:%SZ) -X main.BuildVersion=dev -X main.Commit=$(git rev-parse HEAD)" -o dist/kvx ./cmd/kvx/kvx.go`
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/agents/go-build-resolver.agent.md]: Fixed -- build path updated to . (entrypoint is main.go at repo root). Thanks!

File: .github/instructions/go-conventions.instructions.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+- Bubbletea models implement `Init()`, `Update()`, and `View()` -- keep `Update` focused on state transitions and `View` on rendering
+- Use lipgloss for all terminal styling -- never raw ANSI codes
+- TUI state belongs in `internal/ui/`; reusable library-level TUI entry points in `pkg/tui/`
+
+## Reference
+
+See skill: `golang-patterns` for comprehensive Go idioms and patterns.
+
+See skill: `golang-patterns` for comprehensive Go idioms and patterns.
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/instructions/go-conventions.instructions.md]: Good catch, fixed. Removed the duplicate reference line.

File: .github/prompts/completeness-check.prompt.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+1. Run `git diff --cached --stat` to identify staged changes
+2. If nothing is staged, fall back to `git log origin/main..HEAD --stat` to check pushed commits on the branch
+3. For each feature or command change, verify:
+   - Docs in `docs/`
+   - Tutorials in `docs/tutorials/` for user-facing features
+   - Examples in `examples/`
+   - Unit tests colocated with implementation (`*_test.go`)
+   - Snapshot tests for TUI behavior changes (`--snapshot --press`)
+3. Report present vs missing as a checklist
+4. Do not create anything, just report the gaps
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/prompts/completeness-check.prompt.md]: Good catch, fixed. Renumbered to 4/5.

File: .github/instructions/docs-examples.instructions.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+description: "Documentation and examples conventions for kvx. Use when creating or updating docs, tutorials, or examples."
+applyTo: "**/*.go"
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/instructions/docs-examples.instructions.md]: Good catch, fixed. Updated applyTo to docs/**,examples/**,**/*.md.

File: .github/agents/planner.agent.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
+---
+description: "Feature implementation planner for kvx. Creates structured implementation blueprints with architecture decisions, task breakdown, and dependency analysis. Use for complex features and refactoring."
+name: "planner"
+tools: [vscode/askQuestions, read, search/changes, search/codebase, search/fileSearch, search/listDirectory, search/textSearch, web, agent]
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/agents/planner.agent.md:4]: These tool names (vscode/askQuestions, search/changes, search/codebase, etc.) are valid VS Code Copilot agent tool references -- they map to built-in VS Code capabilities. The other agents use simpler tool sets because they do not need the full search surface. Happy to discuss further.

File: .github/agents/pr-reviewer.agent.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }' -f threadId=<THREAD_ID> -f body="<response>"
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/agents/pr-reviewer.agent.md:108]: Fixed -- updated mutation to addPullRequestReviewThreadReply. Thanks!

File: .github/agents/commit-message.agent.md [hunk 8]
```diff
@@ -0,0 +1,190 @@
+---
+description: "Generates conventional commit messages from staged or recent changes. Analyzes git diff to produce well-structured messages following the project's conventional commits spec. Does NOT execute git commit — only outputs the message. Use when preparing commit messages."
+name: "commit-message"
+tools: [read, execute]
+---
+You are a commit message generator for the **kvx** project. You analyze changes and produce conventional commit messages. You **never** execute `git commit` — you only output the message for the user to copy.
+
+**CRITICAL**: Commit messages feed directly into release notes via GoReleaser and git-cliff. Every commit description appears in the public changelog. Write messages that are **meaningful to users reading a release** — describe the user-facing change, not implementation details.
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/agents/commit-message.agent.md]: Good catch, fixed. Replaced all em dashes with -- for ASCII-only consistency.

File: .github/agents/go-build-resolver.agent.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+```bash
+go build ./...
+go vet ./...
+golangci-lint run
+go mod verify
+```
+
+Only run `go mod tidy -v` **after** fixing module/dependency changes — it is a write operation and should not be part of the initial diagnostic pass.
+
+## Resolution Workflow
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/agents/go-build-resolver.agent.md:33]: Good catch, fixed. Replaced em dashes with --.

File: .github/agents/go-reviewer.agent.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+- **Core engine usage**: Data operations should go through `pkg/core.Engine`, not bypass it
+- **Constants**: No magic strings or numbers — use constants or settings
+- **Error wrapping**: `fmt.Errorf("context: %w", err)` with meaningful context
+- **Logging**: Use `pkg/logger` (logr interface), never `fmt.Println` for debug output
+- **Tests**: Include benchmarks for performance-sensitive code
+
+## Review Priorities
+
+### CRITICAL — Security
+- Command injection: Unvalidated input in `os/exec` or `shellexec`
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/agents/go-reviewer.agent.md:34]: Good catch, fixed. Replaced em dashes with --.

File: .github/agents/issue-creator.agent.md [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+**Feasibility**: Straightforward or blockers/risks?
+
+**Scope**:
+| Size | Description |
+|------|-------------|
+| **XS** | Trivial — config change, typo fix, single-line edit |
+| **S** | Small — isolated change in 1-2 files, < 1 hour |
+| **M** | Medium — touches multiple files/layers, < 1 day |
+| **L** | Large — cross-cutting change, new interfaces, multi-day |
+| **XL** | Extra large — architectural change, major refactor |
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/agents/issue-creator.agent.md]: Good catch, fixed. Replaced em dashes with --.

File: .github/agents/planner.agent.md [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+- Logger: `pkg/logger/` (logr interface with zapr adapter)
+- Settings/config: `pkg/settings/`, `internal/config/`
+- Tests: `testify/assert`, colocated `*_test.go` files, benchmarks for performance-sensitive code
+
+## Planning Process
+
+1. **Understand** — Analyze the request, identify constraints
+2. **Research** — Search the codebase for existing patterns, interfaces, and conventions
+3. **Design** — Create the implementation blueprint
+4. **Review** — Identify risks, edge cases, and dependencies
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/agents/planner.agent.md]: Good catch, fixed. Replaced em dashes with --.

File: .github/prompts/go-test.prompt.md [hunk 13]
```diff
@@ -0,0 +1,13 @@
+---
+description: "kvx: Run Go tests with race detection, check coverage, and diagnose failures via go-build-resolver."
+agent: "go-build-resolver"
+---
+Run the Go test suite and report results:
+
+1. Run `go test -race -count=1 ./...` — report any failures
+2. Run `go test -coverprofile=coverage.out ./...` — check coverage
+3. Run `go tool cover -func=coverage.out | tail -1` — report total coverage
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/prompts/go-test.prompt.md]: Good catch, fixed. Replaced em dashes with --.

File: .github/prompts/go-build.prompt.md [hunk 14]
```diff
@@ -0,0 +1,13 @@
+---
+description: "kvx: Fix Go build errors, go vet warnings, and linter issues with minimal surgical changes."
+agent: "go-build-resolver"
+---
+Diagnose and fix the current Go build errors:
+
+1. Run `go build ./...` to identify compilation errors
+2. Run `go vet ./...` to find vet warnings
+3. Apply minimal, surgical fixes — don't refactor
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/prompts/go-build.prompt.md]: Good catch, fixed. Replaced em dash with --.

File: .github/prompts/plan.prompt.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+Create a structured implementation blueprint for the described feature:
+
+1. **Summary** — What and why
+2. **Architecture decisions** — Layers affected, new types, interface changes
+3. **Task breakdown** — Ordered steps with files, complexity, dependencies
+4. **Interface design** — Define contracts first
+5. **Error handling** — Sentinel errors, wrapping strategy
+6. **Testing strategy** — Unit tests, benchmarks, snapshot tests
+7. **Documentation** — Docs, examples, tutorials
+8. **Risks & edge cases** — What could go wrong
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/prompts/plan.prompt.md]: Good catch, fixed. Replaced em dashes with --.

File: .github/skills/golang-patterns/SKILL.md [hunk 16]
```diff
@@ -0,0 +1,402 @@
...
+- Strings: `maxLength`, `example`, `pattern`, `patternDescription`
+- Integers: `maximum`, `example`
+- Arrays: `maxItems` (no `example`)
+- Objects/maps: no `example` tag
+
+```go
+type Provider struct {
+    Name    string `json:"name" yaml:"name" doc:"Provider name" maxLength:"128" example:"redis"`
+    Version string `json:"version" yaml:"version" doc:"Semantic version" pattern:"^v?\\d+\\.\\d+\\.\\d+$" patternDescription:"semver format" example:"1.0.0"`
+    Port    int    `json:"port" yaml:"port" doc:"Listen port" maximum:"65535" example:"8080"`
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .github/skills/golang-patterns/SKILL.md]: Fixed -- removed Huma-specific validation tag guidance and replaced with generic framework-agnostic advice. Thanks!
